tests/test_job_management.py

```diff
@@ -1,95 +1,253 @@
+                     ))
+
+    # this triggers email
     c = requests.get(server + "/call_back",
-                params={
-                    'job_id': job_id,
-                    'session_id': session_id,
-                    'instrument_name': "empty-async",
-                    'action': 'ready',
-                    'node_id': 'final',
-                    'message': 'done',
-                })
-    
-    print(c.text)    
+                     params=dict(
+                         job_id=job_id,
+                         session_id=session_id,
+                         instrument_name="empty-async",
+                         action='done',
+                         node_id='node_final',
+                         message='done',
+                         token=encoded_token,
+                         time_request=time_request
+                     ))
+
+    job_monitor_call_back_done_json_fn = f'scratch_sid_{session_id}_jid_{job_id}/job_monitor_node_final_done_.json'
+    # the aliased version might have been created
+    job_monitor_call_back_done_json_fn_aliased = f'scratch_sid_{session_id}_jid_{job_id}_aliased/job_monitor_node_final_done_.json'
+    assert os.path.exists(job_monitor_call_back_done_json_fn) or \
+           os.path.exists(job_monitor_call_back_done_json_fn_aliased)
+    assert c.status_code == 200
+    # read the json file
+    if os.path.exists(job_monitor_call_back_done_json_fn):
+        f = open(job_monitor_call_back_done_json_fn)
+    else:
+        f = open(job_monitor_call_back_done_json_fn_aliased)
+
+    jdata = json.load(f)
+    assert jdata['email_status'] == 'email sent'
+
+    # this also triggers email (simulate a failed request)
+    c = requests.get(server + "/call_back",
+                     params={
+                         'job_id': job_id,
+                         'session_id': session_id,
+                         'instrument_name': "empty-async",
+                         'action': 'failed',
+                         'node_id': 'node_failed',
+                         'message': 'failed',
+                         'token': encoded_token,
+                         'time_request': time_request
+                     })
+    job_monitor_call_back_failed_json_fn = f'scratch_sid_{session_id}_jid_{job_id}/job_monitor_node_failed_failed_.json'
+    # the aliased version might have been created
+    job_monitor_call_back_failed_json_fn_aliased = f'scratch_sid_{session_id}_jid_{job_id}_aliased/job_monitor_node_failed_failed_.json'
+
+    assert os.path.exists(job_monitor_call_back_failed_json_fn) or os.path.exists(
+        job_monitor_call_back_failed_json_fn_aliased)
+    assert c.status_code == 200
+    # read the json file
+    if os.path.exists(job_monitor_call_back_failed_json_fn):
+        f = open(job_monitor_call_back_failed_json_fn)
+    else:
+        f = open(job_monitor_call_back_failed_json_fn_aliased)
+
+    jdata = json.load(f)
+    assert jdata['email_status'] == 'email sent'
```

Comment:
could we make the smtp server fixture write the email in a file, read it here and verify it is as expected?

Reply:
It is an option. But it would work only within the specific context of this test ie using a localhost server. In a real context, we don't have the option to inspect log files generated from an SMTP server. 

An alternative would be to check that the email_status was written in the json output file...

Reply:
I mean precisely for the test, that's why I commented on the test code, sorry if I was unclear.

checking email_status would be also useful, but verifying content of the email as delivered to the smtp server, and stored by the test server, is useful, since email is delivered to the user, and needs to be well checked.